platforms/software/dependency-management/src/main/java/org/gradle/internal/artifacts/configurations/NoContextRoleBasedConfigurationCreationRequest.java

```diff
@@ -19,35 +19,36 @@
 import org.gradle.api.internal.artifacts.configurations.ConfigurationRole;
 import org.gradle.api.internal.artifacts.configurations.RoleBasedConfigurationCreationRequest;
 import org.gradle.api.internal.artifacts.configurations.UsageDescriber;
+import org.gradle.api.problems.internal.InternalProblems;
 import org.gradle.internal.deprecation.DeprecatableConfiguration;
 import org.gradle.internal.deprecation.DeprecationLogger;
 
-import java.util.Collections;
-
 /**
  * An {@link AbstractRoleBasedConfigurationCreationRequest} that does not provide any additional contextual
  * information about the request, and emits generic error messages.
  */
 public final class NoContextRoleBasedConfigurationCreationRequest extends AbstractRoleBasedConfigurationCreationRequest {
-    public NoContextRoleBasedConfigurationCreationRequest(String configurationName, ConfigurationRole role) {
-        super(configurationName, role);
+    public NoContextRoleBasedConfigurationCreationRequest(String configurationName, ConfigurationRole role, InternalProblems problemsService) {
+        super(configurationName, role, problemsService);
     }
 
-    private String getUsageDiscoveryMessage(DeprecatableConfiguration conf) {
+    @Override
+    protected String getUsageDiscoveryMessage(DeprecatableConfiguration conf) {
         String currentUsageDesc = UsageDescriber.describeCurrentUsage(conf);
         return String.format("Configuration %s already exists with permitted usage(s):\n" +
             "%s\n", getConfigurationName(), currentUsageDesc);
     }
 
-    private String getUsageExpectationMessage() {
+    @Override
+    protected String getUsageExpectationMessage() {
         String expectedUsageDesc = UsageDescriber.describeRole(getRole());
         return String.format("Yet Gradle expected to create it with the usage(s):\n" +
             "%s\n" +
             "Gradle will mutate the usage of configuration %s to match the expected usage. This may cause unexpected behavior. Creating configurations with reserved names", expectedUsageDesc, getConfigurationName());
     }
 
     @Override
-    public void warnAboutNeedToMutateUsage(DeprecatableConfiguration conf) {
+    public void failOnNeedToMutateUsage(DeprecatableConfiguration conf) {
```

Comment:
This has been renamed to `fail` but does not fail

Reply:
Copy-paste error, good catch.